diffpy/snmf/new_subroutines.py

```diff
@@ -6,8 +6,22 @@
 
 
 def create_components(number_of_components, grid_vector, number_of_signals, signal_length):
+    """Creates the ComponentSignal objects
+
+    Parameters
+    ----------
+    number_of_components: int
+    grid_vector: 1d array
+    number_of_signals: int
+    signal_length: int
```

Comment:
how is this different than the length of grid_vector?  Do you allow the components to have a differnent length than the signals?

Reply:
It is not different. The components should not have a different length than the signals

Reply:
therefore we don't need this as an input because it is `len(grid_vector)`, no?